File: incubator/net.sf.j2s.java.org.eclipse.osgi/src/org/eclipse/core/runtime/adaptor/EclipseElementFactory.java
/*******************************************************************************
 * Copyright (c) 2003, 2005 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package org.eclipse.core.runtime.adaptor;

import java.io.IOException;

import org.eclipse.osgi.framework.adaptor.BundleProtectionDomain;
import org.eclipse.osgi.framework.adaptor.ClassLoaderDelegate;
import org.eclipse.osgi.framework.adaptor.core.*;

/**
 * Creates EclipseBundleData and EclipseClassLoader objects for the framework adaptor
 * <p>
 * Clients may extend this class.
 * </p>
 * @since 3.1
 */
public class EclipseElementFactory implements AdaptorElementFactory {

	public AbstractBundleData createBundleData(AbstractFrameworkAdaptor adaptor, long id) throws IOException {
		return new EclipseBundleData((AbstractFrameworkAdaptor) adaptor, id);
	}

	public org.eclipse.osgi.framework.adaptor.BundleClassLoader createClassLoader(ClassLoaderDelegate delegate, BundleProtectionDomain domain, String[] bundleclasspath, AbstractBundleData data) {
		return new EclipseClassLoader(delegate, domain, bundleclasspath, data.getAdaptor().getBundleClassLoaderParent(), data);
	}

}

File: incubator/net.sf.j2s.test.mauve/j2slib/org/eclipse/core/runtime/adaptor/EclipseElementFactory.js
Clazz.declarePackage ("org.eclipse.core.runtime.adaptor");
Clazz.load (["org.eclipse.osgi.framework.adaptor.core.AdaptorElementFactory"], "org.eclipse.core.runtime.adaptor.EclipseElementFactory", ["org.eclipse.core.runtime.adaptor.EclipseBundleData", "$.EclipseClassLoader"], function () {
c$ = Clazz.declareType (org.eclipse.core.runtime.adaptor, "EclipseElementFactory", null, org.eclipse.osgi.framework.adaptor.core.AdaptorElementFactory);
Clazz.overrideMethod (c$, "createBundleData", 
function (adaptor, id) {
return  new org.eclipse.core.runtime.adaptor.EclipseBundleData (adaptor, id);
}, "org.eclipse.osgi.framework.adaptor.core.AbstractFrameworkAdaptor,~N");
Clazz.overrideMethod (c$, "createClassLoader", 
function (delegate, domain, bundleclasspath, data) {
return  new org.eclipse.core.runtime.adaptor.EclipseClassLoader (delegate, domain, bundleclasspath, data.getAdaptor ().getBundleClassLoaderParent (), data);
}, "org.eclipse.osgi.framework.adaptor.ClassLoaderDelegate,org.eclipse.osgi.framework.adaptor.BundleProtectionDomain,~A,org.eclipse.osgi.framework.adaptor.core.AbstractBundleData");
});
